Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -19,24 +21,106 @@ module.exports.Quantity = class Quantity extends Expression
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value < parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
-    if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
+    if other instanceof Quantity and @unit == other.unit
+      @value == parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration == otherSmallestDuration
+    else null
 
 time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
 
 clean_unit = (units) ->
-  if time_units[units] then time_units[units] else units
+  if ucum_time_units[units] then ucum_to_cql_units[ucum_time_units[units]] else units
+
+# Hash of time units and their UCUM equivalents, both case-sensitive and case-insensitive
+# See http://unitsofmeasure.org/ucum.html#para-31
+# The CQL specification says that dates are based on the Gregorian calendar
+ucum_time_units = {'years': 'a_g', 'year': 'a_g', 'YEARS': 'a_g', 'YEAR': 'a_g', 'a_g': 'a_g'
+  , 'a': 'a_j', 'ANN': 'a_j', 'ann': 'a_j', 'A': 'a_j', 'a_j': 'a_j'
+  , 'months': 'mo_g', 'month':'mo_g', 'mo_g': 'mo_g'
+  , 'mo': 'mo_j', 'MO': 'mo_j', 'mo_j', 'mo_j'
+  , 'weeks': 'wk', 'week': 'wk', 'wk': 'wk', 'WK': 'wk'
+  , 'days': 'd', 'day':'d', 'd': 'd', 'D': 'd'
+  , 'hours': 'h', 'hour': 'h', 'h': 'h', 'H': 'h'
+  , 'minutes': 'min', 'minute': 'min', 'min': 'min', 'MIN': 'min'
+  , 'seconds':'s', 'second':'s', 's': 's', 'S': 's'
+  , 'milliseconds' : 'ms', 'millisecond' : 'ms', 'ms': 'ms', 'MS': 'ms'
+  }
+
+ucum_to_cql_units = {
+    'a_j':  'year'
+  , 'a_g':  'year'
+  , 'mo_j': 'month'
+  , 'mo_g': 'month'
+  , 'wk':   'week'
+  , 'd':    'day'
+  , 'h':    'hour'
+  , 's':    'second'
+  , 'ms':   'millisecond'
+}
+
+get_ucum_units = (units) ->
+  if ucum_time_units[units]
+    ucum_time_units[units]
+  else units
+
+# The smallest common duration is the millisecond
+smallestDuration = (qty) ->
+  if parseFloat qty.value
+    millivalue = switch
+      when get_ucum_unit(qty.unit) == 'second' then qty.value * 1000
+      when get_ucum_unit(qty.unit) == 'minute' then qty.value * 60 * 1000
+      when get_ucum_unit(qty.unit) == 'hour' then qty.value * 60 * 60 * 1000
+      when get_ucum_unit(qty.unit) == 'day' then qty.value * 24 * 60 * 60 * 1000
+      when get_ucum_unit(qty.unit) == 'week' then qty.value * 7 * 24 * 60 * 60 * 1000
```

Comment:
Shouldn't the above 5 lines use the ucum units for comparison (e.g., `s`, `min`, `h`, etc)?